documentation/library_guide/kernel_templates/single_pass_scan.rst

```diff
@@ -0,0 +1,203 @@
++-----------------------------------------------+---------------------------------------------------------------------+
+|                                               | The sequences of elements to apply the algorithm to.                |
+|  - ``in_begin``, ``in_end``, ``out_begin`` (1)| Supported sequence types:                                           |
+|  - ``in_rng``, ``out_rng`` (2)                |                                                                     |
+|                                               | - ``sycl::buffer`` (2),                                             |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::all               |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`oneapi::dpl::experimental::ranges::views::subrange          |
+|                                               |   <viewable-ranges>` (2),                                           |
+|                                               | - :ref:`USM pointers <use-usm>` (1),                                |
+|                                               | - :ref:`oneapi::dpl::begin and oneapi::dpl::end                     |
+|                                               |   <use-buffer-wrappers>` (1).                                       |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``binary_op``                                | A function object that will applied to the elements of the input    |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+|  ``param``                                    | A :doc:`kernel_param <../kernel_configuration>` object.             |
+|                                               |                                                                     |
++-----------------------------------------------+---------------------------------------------------------------------+
+
+
+**Type Requirements**:
+
+- The element type of sequence to scan must be a C++ integral or floating-point type
+  other than ``bool`` with a width of up to 64 bits.
+- The binary operator must have a known identity. That is, ``sycl::has_known_identity<Op>::value`` must
+  evaluate to true.
+- The result is non-deterministic if the binary operator is non-associative (such as in floating-point addition).
+
+.. note::
+
+   Current limitations:
+
+   - The function will internally block until the issued kernels have completed execution
```

Comment:
I added another sentence about how this makes the algorithm synchronous instead of the intent of being asynchronous.

In the future, I'm not sure what the best way is to change the API to show that is now asynchronous. Perhaps it would be better if we change the algorithm now to return `void` and in the future, it will return a `sycl::event`?

Reply:
Its an interesting question about a general mindset for KT, one probably worth discussing, but we can leave it alone in the context of this doc PR.
Thanks for the clarification in the text.